src/compiler/Stage/Parse/Parser.elm

```diff
@@ -378,54 +378,47 @@ literal : Parser_ Frontend.Expr
 literal =
     P.succeed Literal
         |= P.oneOf
-            -- TODO literalFloat
-            [ literalInt
+            [ literalNumber
             , literalChar
             , literalString
             , literalBool
             ]
         |> P.inContext InLiteral
 
 
-literalInt : Parser_ Literal
-literalInt =
+literalNumber : Parser_ Literal
+literalNumber =
     let
-        int =
-            {- This deals with the bug [elm/parser#25](https://github.com/elm/parser/issues/25).
-               In our case it was interfering with `if ... then ... >e<lse ...`.
-
-               For more info see @dmy's assessment at [dmy/elm-pratt-parser#1](https://github.com/dmy/elm-pratt-parser/issues/1).
-            -}
-            P.backtrackable <|
-                P.int ExpectingInt InvalidInt
-
-        hexInt =
-            P.succeed identity
-                |. P.symbol (P.Token "0x" ExpectingHexPrefix)
-                |= P.getChompedString (P.chompWhile Char.isHexDigit)
-                |> P.andThen
-                    (\hexString ->
-                        hexString
-                            -- TODO this String.toLower shouldn't be needed - see https://github.com/rtfeldman/elm-hex/pull/1
-                            |> String.toLower
-                            |> Hex.fromString
-                            |> Result.map P.succeed
-                            |> Result.withDefault (P.problem InvalidHexInt)
-                    )
-                |> P.inContext InHexInt
+        parseLitteralNumber =
+            P.number
+                { int = Ok Int
+                , hex = Ok Int
+                , octal = Ok Int
+                , binary = Ok Int
```

Comment:
Elm 0.19 doesn't support octal and binary numbers, so let's make these two rows `Nothing`